adsingestp/parsers/jats.py

```diff
@@ -852,18 +853,20 @@ def _parse_esources(self):
         links = []
         rawlinks = self.article_meta.find_all("self-uri")
 
-        if rawlinks:
-            for link in rawlinks:
-                if link.get("content-type") == "full_html":
+        for link in rawlinks:
+            if link.get("content-type", "") == "full_html":
+                if validators.url(link.get("xlink:href")):
```

Comment:
To be safe, put an empty string as the default in the `get` here too (and in lines 862 and 869).